### PR TITLE
Docs: deploy-code file updated

### LINKED ISSUE #559: [DOC] - Missing permissions on self hosted bot?
### What is incorrect in the documentation?

I'm not 100% sure if its permission related, but I **think** it is. As my bot doesn't have any permission set except 1:
image

Otherwise, when I run the bot, I get an error with the slash commands:
```
debug: 2021-10-03 18:14:59: registering to home guild only
error: 2021-10-03 18:14:59: There was an error in the slash command register:
error: 2021-10-03 18:14:59: {"errorMessage":"Missing Access","errorStack":"DiscordAPIError[50001]: Missing Access\n    at SequentialHandler.runRequest (/workspaces/discord-bot/node_modules/@discordjs/rest/dist/lib/handlers/SequentialHandler.js:198:23)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async SequentialHandler.queueRequest (/workspaces/discord-bot/node_modules/@discordjs/rest/dist/lib/handlers/SequentialHandler.js:99:20)"}
error: 2021-10-03 18:14:59: failed to register commands.
```

### What should that text say?

Should there be a section in the docs specifying what permissions to set up the bot with at all? Or maybe I just missed it, or my issue when running the bot is totally unrelated?

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by princeprag**
# Pull Request

## Description:

 
The deploy-code file is updated. It now clearly states how to invite the local instance of Becca to the server with appropriate permissions and scopes.

## Related Issue:

Closes #559

### CODE REVIEW

File: docs/deploy-code.md [hunk 1]
```diff
@@ -1,15 +1,30 @@
 # Run Your Version of Becca
 
-The next step is to invite your instance of Becca to your server. You can use the Discord Developer portal to generate an invitation link, or you can use the Client ID from your Becca's application page to create a link yourself. If you wish to manually create the link, the format is:
+The next step is to invite your instance of Becca to your server. You can use the Discord Developer portal to generate an invitation link.
 
-`https://discord.com/oauth2/authorize?client_id=CLIENT_ID_HERE&scope=bot`
+##Generating the link from dashboard in developer portal
```
Comment (nhcarrigan) [inline docs/deploy-code.md:5]:
```suggestion
## Generating the link from dashboard in developer portal
```

File: docs/deploy-code.md [hunk 2]
```diff
@@ -1,15 +1,30 @@
 # Run Your Version of Becca
 
-The next step is to invite your instance of Becca to your server. You can use the Discord Developer portal to generate an invitation link, or you can use the Client ID from your Becca's application page to create a link yourself. If you wish to manually create the link, the format is:
+The next step is to invite your instance of Becca to your server. You can use the Discord Developer portal to generate an invitation link.
```
Comment (nhcarrigan) [inline docs/deploy-code.md:3]:
```suggestion
The next step is to invite your instance of Becca to your server. You can use the Discord Developer portal to generate an invite link.
```

File: docs/deploy-code.md [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
 
-`https://discord.com/oauth2/authorize?client_id=CLIENT_ID_HERE&scope=bot`
+##Generating the link from dashboard in developer portal
 
-Regardless of which method you choose, the link will open a new window allowing you to choose which server you'd like to add Becca to.
+Go to [Application](https://discord.com/developers/applications) in the Discord Developer Portal.
```
Comment (nhcarrigan) [inline docs/deploy-code.md:7]:
```suggestion
Go to the [Applications](https://discord.com/developers/applications) page in the Discord Developer Portal.
```

File: docs/deploy-code.md [hunk 4]
```diff
@@ -1,15 +1,30 @@
...
 
-Once you have added her to your server, switch back to your terminal and run the command npm start. If you have set the `WH_URL` in your environment appropriately, Becca should send a message indicating it is online. Otherwise, you should see the text Activate the Omega! in your terminal, indicating she is online! Now you can try some of the commands to see if she is functioning correctly.
+- Open the new application that you have created in previous step.
```
Comment (nhcarrigan) [inline docs/deploy-code.md:9]:
```suggestion
- Open the new application that you created earlier.
```

File: docs/deploy-code.md [hunk 5]
```diff
@@ -1,15 +1,30 @@
...
+
+- Click on `OAuth2` tab in the left panel.
+
+- In the scopes section click checkboxes with `applications.commands` (you can't run slash commands without it) and `bot`.
```
Comment (nhcarrigan) [inline docs/deploy-code.md:13]:
```suggestion
- In the scopes section, select the checkboxes with `applications.commands` (allows your bot to register slash commands) and `bot`.
```

File: docs/deploy-code.md [hunk 6]
```diff
@@ -1,15 +1,30 @@
...
+
+- In the bot permissions section click on checkboxes as stated in [set-permissions](./set-permissions.md) file.
+
```
Comment (nhcarrigan) [inline docs/deploy-code.md:16]:
```suggestion

Your choices should look like this:

![image of choices list](https://cdn.nhcarrigan.com/content/discord/becca-permissions.png)

```

Comment (nhcarrigan) [reply docs/deploy-code.md:16]: Uploaded an image to our CDN for this. 🙂

File: docs/deploy-code.md [hunk 7]
```diff
@@ -1,15 +1,30 @@
...
+You can then copy the invitation link from the scopes section above using the `Copy` button.
```
Comment (nhcarrigan) [inline docs/deploy-code.md:17]:
```suggestion
You can then copy the invite link from the scopes section above using the `Copy` button.

```

File: docs/deploy-code.md [hunk 8]
```diff
@@ -1,15 +1,30 @@
...
+>It looks something like
+> `https://discord.com/oauth2/authorize?client_id=CLIENT_ID_HERE&permissions=INTEGER_CODE_HERE&scope=applications.commands%20bot`
+
```
Comment (nhcarrigan) [inline docs/deploy-code.md:20]:
````suggestion
It looks something like: 

```txt
https://discord.com/oauth2/authorize?client_id=CLIENT_ID_HERE&permissions=INTEGER_CODE_HERE&scope=applications.commands%20bot
```
````

Block quote format doesn't work here. Also, let's use a multi-line code block so we get horizontal scroll on the link.

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
This is off to a great start! Some grammar/verbiage tweaks, and I've got an image we can use for the permissions.

**Comment by princeprag** [comment]
@nhcarrigan Changes done!

**Review by nhcarrigan — APPROVED** [review]
Looks good to me!

**Pull request merged**